src/start_server.ts

```diff
@@ -86,7 +85,7 @@ async function applyDefaultOptions(options: ServerOptions):
     Promise<ServerOptions> {
       const withDefaults = Object.assign({}, options);
       Object.assign(withDefaults, {
-        port: await nextOpenPort(options.port),
+        port: options.port || 0,
```

Comment:
This function could be sync now. Probably being async to choose a port wasn't great to begin with.